src/ensembl/src/content/app/species/state/general/speciesGeneralHelper.ts

```diff
@@ -0,0 +1,462 @@
+  [Stats.STRAIN]: { label: 'Breed/Cultivar/Strain' },
+  [Stats.SEX]: { label: 'Sex' },
+  [Stats.SCIENTIFIC_NAME]: { label: 'Scientific name' },
+  [Stats.PSEUDOGENES]: { label: 'Pseudogenes' }
+};
+
+type IndividualStat = Omit<
+  SpeciesStatsProps,
+  'primaryValue' | 'secondaryValue'
+> & {
+  primaryValue: string | number | null;
+  secondaryValue?: string | number | null;
+};
+
+type StatsGroup = {
+  title: string;
+  stats: [IndividualStat[]];
+};
+
+export type StatsSection = {
+  section: SpeciesStatsSection;
+  exampleLinks?: Partial<urlObj>;
+  primaryStats?: IndividualStat;
+  secondaryStats?: IndividualStat;
+  groups: StatsGroup[];
+};
+
+type BuildStatProps = Partial<IndividualStat> & {
+  primaryKey: Stats;
+  secondaryKey?: Stats;
+};
+
+const buildIndividualStat = (
+  props: BuildStatProps
+): IndividualStat | undefined => {
+  let primaryValue = props.primaryValue;
+
+  if (!primaryValue) {
+    return;
+  }
+
+  const formattingOptions = statsFormattingOptions[props.primaryKey];
+
+  if (!isNaN(Number(primaryValue))) {
+    primaryValue = getCommaSeparatedNumber(Number(primaryValue));
+  }
+
+  return {
+    label: formattingOptions?.label || props.primaryKey,
+    primaryValue,
+    primaryUnit: formattingOptions?.primaryUnit,
+    primaryValuePostfix: formattingOptions?.primaryValuePostfix
+  };
+};
+
+const buildHeaderStat = (props: BuildStatProps): IndividualStat | undefined => {
+  let primaryValue = props.primaryValue;
+
+  if (!primaryValue) {
+    return;
+  }
+
+  const formattingOptions = statsFormattingOptions[props.primaryKey];
+
+  if (!isNaN(Number(primaryValue))) {
+    primaryValue = getCommaSeparatedNumber(Number(primaryValue));
+  }
+
+  return {
+    label: props.primaryKey,
+    primaryValue,
+    primaryUnit: formattingOptions?.headerUnit
+  };
+};
+
+const getExampleLinks = (props: {
+  section: SpeciesStatsSection;
+  genome_id: string;
+  exampleFocusObjects: ExampleFocusObject[];
+}) => {
+  const { section, genome_id, exampleFocusObjects } = props;
+
+  const exampleLinks: Partial<urlObj> = {};
+
+  if (section === SpeciesStatsSection.CODING_STATS) {
+    const geneExample = exampleFocusObjects.find(
+      (object) => object.type === 'gene'
+    );
+
+    const focusId = geneExample?.id
+      ? buildFocusIdForUrl({
+          type: 'gene',
+          objectId: geneExample?.id
+        })
+      : undefined;
+
+    if (focusId) {
+      exampleLinks.genomeBrowser = urlFor.browser({
+        genomeId: genome_id,
+        focus: focusId
+      });
+
+      exampleLinks.entityViewer = urlFor.entityViewer({
+        genomeId: genome_id,
+        entityId: focusId
+      });
+    }
+  } else if (section === SpeciesStatsSection.ASSEMBLY_STATS) {
+    const regionExample = exampleFocusObjects.find(
+      (object) => object.type === 'region'
+    );
+
+    const focusId = regionExample?.id
+      ? buildFocusIdForUrl({
+          type: 'region',
+          objectId: regionExample?.id
```

Comment:
at this point, `regionExample` definitely exists (same above with `geneExample`).

Reply:
Updated it